src/repart/repart.c

```diff
@@ -678,6 +695,9 @@ static void partition_foreignize(Partition *p) {
         p->compression = mfree(p->compression);
         p->compression_level = mfree(p->compression_level);
 
+        copy_files_line_free_many(p->copy_files, p->n_copy_files);
+        p->copy_files = NULL;
```

Comment:
also set n_copy_files to zero

Reply:
...and copy_files to NULL, by that logic.  Added both.